src/v2i-hub/MUSTSensorDriverPlugin/src/MUSTSensorDriverPlugin.cpp

```diff
@@ -0,0 +1,116 @@
+#include "MUSTSensorDriverPlugin.h"
+
+using namespace tmx::utils;
+using namespace std;
+
+namespace MUSTSensorDriverPlugin {
+	
+	MUSTSensorDriverPlugin::MUSTSensorDriverPlugin(const string &name): PluginClientClockAware(name)
+	{
+		mustSensorPacketReceiverThread = std::make_unique<tmx::utils::ThreadTimer>(std::chrono::milliseconds(5));
+		if (PluginClientClockAware::isSimulationMode()) {
+			PLOG(tmx::utils::logINFO) << "Simulation mode on " << std::endl;
+			SubscribeToMessages();
+		}
+	}
+
+	void  MUSTSensorDriverPlugin::OnStateChange(IvpPluginState state) {
+		PluginClientClockAware::OnStateChange(state);
+		if (state == IvpPluginState_registered) {
+			UpdateConfigSettings();
+		}else {
+			connected = false;
+			SetStatus(keyMUSTSensorConnectionStatus, "DISCONNECTED");
+		}
+	}
+
+	void MUSTSensorDriverPlugin::UpdateConfigSettings()
+	{
+		// Configuration settings are retrieved from the API using the GetConfigValue template class.
+		// This method does NOT execute in the main thread, so variables must be protected
+		// (e.g. using std::atomic, std::mutex, etc.).
+		if (this->IsPluginState(IvpPluginState_registered)) {
+			std::scoped_lock<std::mutex> lock(_configMutex);
+			GetConfigValue<std::string>("ProjectionString", projString);
+			GetConfigValue<std::string>("SensorId", sensorId);
+			// Setup  UDP Server 
+			std::string ip_address;
+			unsigned int port;
+			GetConfigValue<std::string>("DetectionReceiverIP", ip_address);
+			GetConfigValue<uint>("DetectionReceiverPort", port);
+			createUdpServer(ip_address, port);
+			SetStatus(keyMUSTSensorConnectionStatus, "IDLE");
+
+			mustSensorPacketReceiverThreadId = mustSensorPacketReceiverThread->AddPeriodicTick([this]() {
+            	this->processMUSTSensorDetection();
+
+       		 	} // end of lambda expression
+        		, std::chrono::milliseconds(5) );
+        	mustSensorPacketReceiverThread->Start();
+		}
+	}
+	void MUSTSensorDriverPlugin::processMUSTSensorDetection(){
+		if (mustSensorPacketReceiver) {
+			try {
+				MUSTSensorDetection detection = csvToDectection(mustSensorPacketReceiver->stringTimedReceive());
+				if ( !connected ) {
+					connected = true;
+					SetStatus(keyMUSTSensorConnectionStatus, "CONNECTED");
+				}
+				tmx::messages::SensorDetectedObject msg = mustDetectionToSensorDetectedObject(detection, sensorId, projString);
+				PLOG(logDEBUG1) << "Sending Simulated SensorDetectedObject Message " << msg << std::endl;
+				this->BroadcastMessage<tmx::messages::SensorDetectedObject>(msg, _name, 0 , IvpMsgFlags_None);
+			}
+			catch( const tmx::utils::UdpServerRuntimeError &e) {
+				PLOG(logERROR) << "Error occurred processing MUSTSensorDetection" << e << std::endl;
+				SetStatus(keyMUSTSensorConnectionStatus, "DISCONNECTED");
+				connected = false;
+			}
+			catch ( const tmx::TmxException &e){
+				PLOG(logERROR) << "Error occurred processing MUSTSensorDetection" << e.what() << std::endl;
+				SetStatus(keyMUSTSensorConnectionStatus, "DISCONNECTED");
```

Comment:
Rather than repeatedly calling SetStatus(keyMUSTSensorConnectionStatus, "DISCONNECTED"); maybe move it to the end. If not connected, set it to "DISCONNECTED"

Reply:
No the exception would trigger connection to change to false and status to be updated